math/src/elliptic_curve/element.rs

```diff
@@ -1,19 +1,17 @@
-use crate::cyclic_group::IsCyclicBilinearGroup;
-use crate::elliptic_curve::traits::{HasDistortionMap, HasEllipticCurveOperations};
+use crate::cyclic_group::IsCyclicGroup;
+use crate::elliptic_curve::traits::IsEllipticCurve;
 use crate::field::element::FieldElement;
 use std::fmt::Debug;
-use std::marker::PhantomData;
 
 /// Represents an elliptic curve point using the projective short Weierstrass form:
 /// y^2 * z = x^3 + a * x * z^2 + b * z^3,
 /// where `x`, `y` and `z` variables are field elements.
 #[derive(Debug, Clone)]
-pub struct EllipticCurveElement<E: HasEllipticCurveOperations> {
-    value: [FieldElement<E::BaseField>; 3],
-    elliptic_curve: PhantomData<E>,
+pub struct EllipticCurveElement<E: IsEllipticCurve> {
+    pub value: [FieldElement<E::BaseField>; 3],
 }
 
-impl<E: HasEllipticCurveOperations> EllipticCurveElement<E> {
+impl<E: IsEllipticCurve> EllipticCurveElement<E> {
     /// Creates an elliptic curve point giving the projective [x: y: z] coordinates.
     pub fn new(value: [FieldElement<E::BaseField>; 3]) -> Self {
         assert_eq!(
```

Comment:
This function should return a `Result`instead of using an `assert` because doing the latter would panic in runtime. 

In general, if you don't have any choice but to panic (this is not the case tho) it is better to describe the cases in which that function would panic and why in the function documentation.